pkg/api/healthprobe/healthprobe.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package healthprobe
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const defaultTimeout = time.Second
+
+// Serve configures and starts the http server for the health check.
+// It returns an error if the setup failed, or runs the server in a goroutine.
+// Stop the server by cancelling the passed context.
+func Serve(ctx context.Context, port int) error {
```

Comment:
Might be a stupid question, but do we really need yet another server? Is there no way we could do this over https (I might be mistaken but I believe it's supported) with the existing server (@ 5001)?

Reply:
IIRC the current `5001` api handler only listens on localhost, while this healthcheck will be a request from the kubelet (using the host's IP) to the container.
Opening port 5001 on all interfaces could have security implications, this is why I went with a dedicated listener. An alternative would be to open 5001 to all interfaces, but rework the security logic:
  - for the `/health` endpoint, accept all requests, with no auth
  - else, check origin IP and token
WDYT?

Port usage when using containers is not as much of an issue, as the agent pod owns its own IP. But indeed, that's why I went with disabling it by default, to avoid using another port when not useful.